api/v1alpha1/clienttrafficpolicy_types.go

```diff
@@ -136,6 +136,12 @@ type HeaderSettings struct {
 	// +optional
 	PreserveXRequestID *bool `json:"preserveXRequestID,omitempty"`
 
+	// GenerateRequestID configures whether envoy will generate the x-request-id header if it does not exist.
+	// Generating a random UUID4 is expensive so in high throughput scenarios where this feature is not desired it can be disabled.
+	// This feature is enabled by default when empty
+	// +optional
+	GenerateRequestID *bool `json:"generateRequestID,omitempty"`
```

Comment:
thoughts on `DisableRequestID` ?
cc @envoyproxy/gateway-maintainers

Reply:
I'm fine with the name, but this should be opt-in, and didn't change previous behavior(even it's not quite right).

Reply:
This wouldn't change existing behavior. By default Envoy enables this feature when left omitted and is the same approach taken here.
* Explicit enable - adds the field to the config but effectively a no-op
* Explicit disable - disables the feature
* Omitted - no-op for the resulting envoy config and is also omitted there

Reply:
IIRC, there're 200+ files changed before, that's why I call it out.

Reply:
3 options here
```
a) GenerateRequestId
b) DisableRequestId
c) RequestId: "Preserve" | "Overwrite" | "Disable"
```
cc @envoyproxy/gateway-reviewers

Reply:
+1 on `c) RequestId: "Preserve" | "Overwrite" | "Disable"`

Reply:
I think we may four options but I like that approach. What behavior would `Overwrite` lead to there?

What about:
```
a) "PreserveOrGenerate" (preserve if exists, generate otherwise) # naming could be better here
b) "Preserve" (preserve if exists, do not generate)
c) "Generate" (ignore external, always generate)
d) "Disabled" (do not preserve, do not generate)
```

Reply:
in that case, option (c) is better choice.

Reply:
Updated with the new field and a deprecation notice for PreserveXRequestID. I'll mark this as resolved.